ext/dba/dba.c

```diff
@@ -262,9 +262,11 @@ static void dba_close(dba_info *info)
 	if (info->hnd) {
 		info->hnd->close(info);
 	}
-	if (info->path) {
-		pefree(info->path, info->flags&DBA_PERSISTENT);
-	}
+	ZEND_ASSERT(info->path);
+	// Cannot use zend_string_release_ex(info->path, info->flags&DBA_PERSISTENT); as this fails GC assertion?
+	zend_string_free(info->path);
+	info->path = NULL;
```

Comment:
Which test is failing the assertion?

If the failing assertion is `ZEND_ASSERT(GC_FLAGS(s) & IS_STR_PERSISTENT);`, there is a discrepancy between the persistent flag in `info->flags` and whether `info->path` is actually persistent.

`zend_string_release()` (non _ex variant) would be the right function to use here. `zend_string_release_ex()` is an optimization for when the persistent argument is known at compile time (removes a branch).

`zend_string_free()` is also suitable, as long as refcount is 1, but this seems less maintainable.

Reply:
The failling assertion is:
```
php: /home/girgias/Dev/php-src/Zend/zend_rc_debug.c:38: void ZEND_RC_MOD_CHECK(const zend_refcounted_h *): Assertion `(zval_gc_flags(p->u.type_info) & ((1<<7)|(1<<8))) != (1<<7)' failed.
```

Reply:
Ah, thank you! I didn't have the ZEND_RC_DEBUG flag.

ZEND_RC_MOD_CHECK is complaining that we should not touch the refcount of a PERSISTENT string, under the expectation that PERSISTENT strings are not refcounted.

We can disable this check by using `GC_MAKE_PERSISTENT_LOCAL(s)` in `php_dba_zend_string_dup_safe` when persistent is true. 

We also need to change `php_dba_zend_string_dup_safe` so that PERSISTENT strings are duplicated.

Reply:
Thank you for the explanations! Will do this now. :)